src/densities.jl

```diff
@@ -84,12 +84,16 @@ end
         (ik, n) = kn
 
         kpt = basis.kpoints[ik]
-        ifft!(storage.ψnk_real, basis, kpt, ψ[ik][:, n])
+        if isnothing(ψ_real)
+            ifft!(storage.ψnk_real, basis, kpt, ψ[ik][:, n])
+        else
+            storage.ψnk_real .= ψ_real[ik][:, :, :, n]
```

Comment:
you can just avoid this copy by havnig a psink_real which points either to storage.psink_real or to psi_real